appinventor/components/src/com/google/appinventor/components/runtime/DataCollection.java

```diff
@@ -0,0 +1,1047 @@
+      + "x value. A single entry is represented as a List of values of the entry.")
+  public YailList GetEntriesWithXValue(final String x) {
+    try {
+      return threadRunner.submit(new Callable<YailList>() {
+        @Override
+        public YailList call() {
+          // Use X Value as criterion to filter entries
+          return dataModel.findEntriesByCriterion(x, ChartDataModel.EntryCriterion.XValue);
+        }
+      }).get();
+    } catch (InterruptedException e) {
+      Log.e(this.getClass().getName(), e.getMessage());
+    } catch (ExecutionException e) {
+      Log.e(this.getClass().getName(), e.getMessage());
+    }
+
+    // Undefined behavior: return empty List
+    return new YailList();
+  }
+
+  /**
+   * Returns all entries of the data series matching the specified y value.
+   * For a description of the format of the returned List, see {@link #GetAllEntries()}
+   *
+   * @param y y value to search for
+   * @return YailList of entries (represented as tuples)
+   */
+  @SuppressWarnings("TryWithIdenticalCatches")
+  @SimpleFunction(description = "Returns a List of entries with y values matching the specified "
+      + "y value. A single entry is represented as a List of values of the entry.")
+  public YailList GetEntriesWithYValue(final String y) {
+    try {
+      return threadRunner.submit(new Callable<YailList>() {
+        @Override
+        public YailList call() {
+          // Use YValue as criterion to filter entries
+          return dataModel.findEntriesByCriterion(y, ChartDataModel.EntryCriterion.YValue);
+        }
+      }).get();
+    } catch (InterruptedException e) {
+      Log.e(this.getClass().getName(), e.getMessage());
+    } catch (ExecutionException e) {
+      Log.e(this.getClass().getName(), e.getMessage());
+    }
+
+    return new YailList();
+  }
+
+  /**
+   * Returns all entries of the data series.
+   * The returned value is a list, where each element of the list
+   * is a list containing the values of the entry in order.
+   *
+   * @return YailList of all the entries of the Data Series
+   */
+  @SuppressWarnings("TryWithIdenticalCatches")
+  @SimpleFunction(description = "Returns all the entries of the Data Series. "
+      + "A single entry is represented as a List of values of the entry.")
+  public YailList GetAllEntries() {
+    try {
+      return threadRunner.submit(new Callable<YailList>() {
+        @Override
+        public YailList call() {
+          return dataModel.getEntriesAsTuples();
+        }
+      }).get();
+    } catch (InterruptedException e) {
+      Log.e(this.getClass().getName(), e.getMessage());
+    } catch (ExecutionException e) {
+      Log.e(this.getClass().getName(), e.getMessage());
+    }
+
+    return new YailList();
+  }
+
+  /**
+   * Imports data from the specified TinyDB component by taking the value
+   * identified by the specified tag value.
+   * <p>
```

Comment:
Don't use `<p>` tags in the Javadocs for block methods. Instead, indent the following paragraph with two spaces.